### PR TITLE
chore: give context to errors

### PR DESCRIPTION
**Pull request by ashinsabu3**
related to #10592

A few more have been fixed

### CODE REVIEW

File: applicationset/generators/merge.go [hunk 1]
```diff
@@ -41,7 +41,7 @@ func (m *MergeGenerator) getParamSetsForAllGenerators(generators []argoprojiov1a
 	for _, generator := range generators {
 		generatorParamSets, err := m.getParams(generator, appSet)
 		if err != nil {
-			return nil, err
+			return nil, fmt.Errorf("error getting params from generator: %w", err)
```
Comment (crenshaw-dev) [inline applicationset/generators/merge.go:44]: How about getting just slightly fancier?

```suggestion
			return nil, fmt.Errorf("error getting params from generator %d of %d: %w", i+1, len(generators), err)
```

File: reposerver/repository/repository.go [hunk 2]
```diff
@@ -1102,15 +1102,15 @@ func helmTemplate(appPath string, repoRoot string, env *v1alpha1.Env, q *apiclie
...
+			return nil, fmt.Errorf("error resolving helm value files: %w", err)
 		}
 
 		templateOpts.Values = resolvedValueFiles
 
 		if !appHelm.ValuesIsEmpty() {
 			rand, err := uuid.NewRandom()
 			if err != nil {
-				return nil, err
+				return nil, fmt.Errorf("error generating random unique id: %w", err)
```
Comment (crenshaw-dev) [inline reposerver/repository/repository.go:1113]:
```suggestion
				return nil, fmt.Errorf("error generating random filename for Helm values file: %w", err)
```

File: reposerver/repository/repository.go [hunk 3]
```diff
@@ -1218,13 +1218,13 @@ func getResolvedValueFiles(
 			// If the $-prefixed path appears to reference another source, do env substitution _after_ resolving that source.
 			resolvedPath, err = getResolvedRefValueFile(rawValueFile, env, allowedValueFilesSchemas, referencedSource.Repo.Repo, gitRepoPaths)
 			if err != nil {
-				return nil, err
+				return nil, fmt.Errorf("error getting value file: %w", err)
```
Comment (crenshaw-dev) [inline reposerver/repository/repository.go:1221]:
```suggestion
				return nil, fmt.Errorf("error resolving value file path: %w", err)
```

File: reposerver/repository/repository.go [hunk 4]
```diff
@@ -1505,7 +1505,7 @@ func GetAppSourceType(ctx context.Context, source *v1alpha1.ApplicationSource, a
 	}
 	appType, err := discovery.AppType(ctx, appPath, repoPath, enableGenerateManifests, tarExcludedGlobs)
 	if err != nil {
-		return "", err
+		return "", fmt.Errorf("error getting application type string: %v", err)
```
Comment (crenshaw-dev) [inline reposerver/repository/repository.go:1508]:
```suggestion
		return "", fmt.Errorf("error getting app source type: %v", err)
```

**Review by crenshaw-dev — CHANGES_REQUESTED** [review]
Thanks! Just a few small tweaks

File: applicationset/generators/merge.go (hunk 1 above)
Comment (ashinsabu3) [reply applicationset/generators/merge.go:44]: makes sense, its definitely better to give more context

Comment (crenshaw-dev) [reply applicationset/generators/merge.go:44]: Yep!

**Review by crenshaw-dev — APPROVED** [review]
Thanks, @ashinsabu3!

**Comment by ashinsabu3** [comment]
There's a unit test which needs to be changed to accomodate the new error message, should I change it? @crenshaw-dev

**Comment by ashinsabu3** [comment]
Seems like those are all the unit tests that needed changes.

**Comment by ashinsabu3** [comment]
If any changes are needed in the unit tests themselves other than this, lmk @crenshaw-dev

**Pull request merged**